### PR TITLE
Update explode to handle lists

### PR DESCRIPTION
**Pull request by rchiodo**
Based on this feedback here that we received: 
https://github.com/microsoft/pylance-release/issues/3871

It looks like explode also takes a list.

- [x] Closes https://github.com/microsoft/pylance-release/issues/3871
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by bashtage** [comment]
Since str is included, why not Sequence[Hashable]

**Comment by rchiodo** [comment]
> Since str is included, why not Sequence[Hashable]

Good idea, I'll switch it to that.

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -1011,7 +1011,9 @@ class DataFrame(NDFrame, OpsMixin):
     def stack(
         self, level: Level | list[Level] = ..., dropna: _bool = ...
     ) -> DataFrame | Series[Any]: ...
-    def explode(self, column: _str | tuple, ignore_index: _bool = ...) -> DataFrame: ...
+    def explode(
+        self, column: _str | tuple | Sequence[Hashable], ignore_index: _bool = ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:1015]: You can just use `Sequence[Hashable]` by itself, as that will match `_str` and `tuple` as well.

**Review by Dr-Irv — APPROVED** [review]
thanks @rchiodo

**Pull request merged**